### PR TITLE
 Adding isPermissionExist funtion  to permission provider.

### PR DESCRIPTION
**Pull request by Yasara123**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/PermissionProvider.java [hunk 1]
```diff
@@ -37,6 +37,13 @@
      */
     void addPermission(Permission permission) throws PermissionException;
 
+    /**
+     * Check permission.
+     *
+     * @param permission
```
Comment (gokul) [inline components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/PermissionProvider.java]: Can you please explain what the parameter is (i.e. in Javadoc comments)

File: components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java [hunk 2]
```diff
@@ -81,6 +81,38 @@ public void addPermission(Permission permission) {
         }
     }
 
+    /**
+     * Check Permission already there.
+     *
+     * @param permission
```
Comment (gokul) [inline components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java]: Please explain the parameter

File: components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java [hunk 3]
```diff
@@ -81,6 +81,38 @@ public void addPermission(Permission permission) {
...
+            ps.setString(2, permission.getPermissionString());
+            if (log.isDebugEnabled()) {
+                log.debug("Executing query: " + query);
+            }
+            resultSet = ps.executeQuery();
+            while (resultSet.next()) {
+                hasPermission = true;
+            }
+        } catch (SQLException e) {
+            throw new PermissionException("Unable to check permissions exist. [Query=" + query + "]", e);
```
Comment (gokul) [inline components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java]: We are just printing the query template we used as a basis of the PreparedStatement here (with the placeholder "?" values. Is this needed?

File: components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/PermissionProvider.java (hunk 1 above)
Comment (Yasara123) [reply components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/PermissionProvider.java]: Fixed 4911621

File: components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java (hunk 2 above)
Comment (Yasara123) [reply components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java]: Fixed 4911621

File: components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java (hunk 3 above)
Comment (Yasara123) [reply components/permission-provider/org.wso2.carbon.analytics.permissions/src/main/java/org/wso2/carbon/analytics/permissions/internal/dao/PermissionsDAO.java]: Fixed 4911621

**Pull request merged**